### DISCUSSION TITLE
Prevent up.submit() from re-enabling submit button

### DISCUSSION THREAD

**Discussion by apokaliptis** [unpoly/unpoly#459] [Q&A]
I'm working on improving my contact form, and one of the things I'm trying to do is prevent the user from clicking the submit button again while the form is still processing. All I have to do is disabled after they hit submit and re-enable it once it is done. I noticed, however, that Unpoly appears to be re-enabling the submit button before I want it to be when I use up.submit(). Why is it it doing that and is there a way to stop it?

**Comment by triskweline** [comment]
Unpoly 2 does not disable or enable submit buttons anywhere. To help other people help you debug, you should prepare a [Glitch project](https://glitch.com/edit/#!/empty-unpoly-project) that reproduces your issue with minimal code.

The upcoming [Unpoly 3](https://github.com/unpoly/unpoly/discussions/407) has an `[up-feedback]` attribute (or `{ feedback }`) option to [disable forms while submitting](http://triskweline.de/unpoly3-slides/#33).

**Reply by apokaliptis** [reply]
I'm not sure why it was re-enabling, but I ended up just swapping the button with a loading spinner while the form is submitting, which looks more elegant.
